Fix for Halloween update

developer9998 opened the pull request:
This fixes hats from not working as well as a few other issues that were in the mod.

![image](https://user-images.githubusercontent.com/81720436/196537994-4b8f293e-33b1-48ae-9735-fc3012b8077a.png)

Preview video: https://www.youtube.com/watch?v=jl8esnwWxPw&ab_channel=dev%27smainchannel.

UPDATE: The button bug has been fixed in the latest commit

developer9998 committed:
Fix for Paintbrawl update

-Updated CosmeticItemUtils.cs
-Updated SelectionManager.cs

Graicc commented:
> Not sure if it was in the prior update, but your original cosmetic will still be equipped if you have a modded cosmetic when the game is open

You changed the method responsible for removing the original cosmetic

Graicc committed:
Check all items for being a hat

developer9998 committed:
summary

i dont know what to call this summary so it's just summary.

haven't fully tested this (i know.) i will as soon as i commit this

also, i know this isnt the best way i could've fixed this just like the last two "disasters" but it still works (i think)

developer9998 committed:
fixed game cosmetic sometimes not resetting

hi

developer9998 committed:
i forgor 💀

if you were switching from modded cosmetic to game cosmetic the button wouldn't change to red because the game thinks you dont have the cosmetic on

Graicc commented on GorillaCosmetics/UI/SelectionManager.cs:
```diff
@@ -405,22 +424,62 @@ public void ResetMaterial()
+            if (PhotonNetwork.InRoom)
+            {
+                if (onlineCustomCosmeticsController == null)
+                    onlineCustomCosmeticsController ??= GorillaTagger.Instance?.myVRRig?.GetComponent<ICustomCosmeticsController>(); // TODO: ew
```
`??=` does the null check implicitly

`onlineCustomCosmeticsController ??= GorillaTagger.Instance?.myVRRig?.GetComponent<ICustomCosmeticsController>();` really should only exist in one place, like how it was in the previous implementation

developer9998 committed:
fixed an issue with the game buttons

finally lol
if you had a modded hat and tried to put on a game hat the button wouldn't change it's appearance

developer9998 commented:
still having issues with onlineCustomCosmeticsController and this error happens when you join a lobby sometimes: 
`[Error  : Unity Log] Error while updating player cosmetics for (USERNAME): System.NullReferenceException: Object reference not set to an instance of an object
  at GorillaCosmetics.CosmeticsNetworker.OnPlayerPropertiesUpdate (Photon.Realtime.Player targetPlayer, ExitGames.Client.Photon.Hashtable changedProps) [0x00019] in <68979d7e9d364e8a99bf91481e1dcd61>:0`

developer9998 committed:
hi

developer9998 committed:
Fix button material

developer9998 committed:
Null check

developer9998 committed:
Fixed mirror

developer9998 committed:
Minor adjustments

- Switch to local Valve.Newtonsoft.Json.dll location (Probably a bad idea lmao)
- Adjusted colour system to ensure both Custom Colour functionality and to see if the option is enabled

developer9998 commented:
Forgot to mention turns out latest commit fixes a lot of custom colour errors by (for the most part) changing up one line lmao

sonc9r9843724839 commented:
fork fork where is it

developer9998 committed:
Update CosmeticsNetworker.cs

- Made it so it will now use the player's full custom properties instead of the changed ones. This is so if another CustomProperties network based mod is used the hats wont reset. Atleast that's what I think may be going on

developer9998 committed:
Nerermind

developer9998 committed:
Fix for game update

Probably not perfect, but it again now fully works
